### PR TITLE
Add custom NitroEnclaveVerifier contract and update multiproof contracts

### PR DESCRIPTION
**Pull request by roger-bai-coinbase**
- Add NitroEnclaveVerifier with ZK-based AWS Nitro attestation verification
- Add tests for NitroEnclaveVerifier
- Add proof submitter configurability to NitroEnclaveVerifier
- Update TEEVerifier, MockVerifier, and deploy scripts accordingly
- Update snapshots and semver-lock

Made-with: Cursor

### CODE REVIEW

File: src/multiproof/tee/NitroEnclaveVerifier.sol [hunk 1]
```diff
@@ -0,0 +1,660 @@
...
+
+        _zkVerifierRoutes[_zkCoProcessor][_selector] = FROZEN;
+        emit ZkRouteWasFrozen(_zkCoProcessor, _selector);
+    }
+
+    /**
+     * @dev Sets the proof submitter address
+     * @param _proofSubmitter The address of the proof submitter
+     */
+    function setProofSubmitter(address _proofSubmitter) external onlyOwner {
```
Comment (leopoldjoy) [inline src/multiproof/tee/NitroEnclaveVerifier.sol:518]: `setProofSubmitter` allows `address(0)`, which would effectively brick `verify()` and `batchVerify()` — no one can have msg.sender == address(0). Consider adding: `require(_proofSubmitter != address(0), "Proof submitter cannot be zero address");`

File: src/multiproof/tee/NitroEnclaveVerifier.sol [hunk 2]
```diff
@@ -0,0 +1,660 @@
...
+    address public proofSubmitter;
+
+    /// @dev Configuration mapping for each supported ZK coprocessor type
+    mapping(ZkCoProcessorType => ZkCoProcessorConfig) public zkConfig;
+
+    /// @dev Mapping of trusted intermediate certificate hashes (excludes root certificate)
+    mapping(bytes32 trustedCertHash => bool) public trustedIntermediateCerts;
+
+    /// @dev Maximum allowed time difference in seconds for attestation timestamp validation
+    uint64 public maxTimeDiff;
```
Comment (leopoldjoy) [inline src/multiproof/tee/NitroEnclaveVerifier.sol:65]: Not necessarily actionable, but currently there's no way to update `maxTimeDiff` after construction. So if operational requirements change (e.g., need to tighten or relax the time window), the contract must be fully redeployed. I suppose this is fine, but wanted to highlight this decision.

File: src/multiproof/tee/NitroEnclaveVerifier.sol [hunk 3]
```diff
@@ -0,0 +1,660 @@
...
+     */
+    function verify(
+        bytes calldata output,
+        ZkCoProcessorType zkCoprocessor,
+        bytes calldata proofBytes
+    )
+        external
+        returns (VerifierJournal memory journal)
+    {
+        require(msg.sender == proofSubmitter, "Only the proof submitter can verify proofs");
```
Comment (leopoldjoy) [inline src/multiproof/tee/NitroEnclaveVerifier.sol]: I noticed that the contract mixes string reverts (`require(msg.sender == proofSubmitter, "Only the proof submitter can verify proofs")`, `revert("Not implemented")`, `revert("Certificate not found...")`) with custom errors (`revert Unknown_Zk_Coprocessor()`, `revert ZkRouteFrozen(...)`). String reverts cost significantly more gas. Since the interface already defines custom errors, consider adding new custom errors for the custom logic (e.g., `error Unauthorized()`, etc.) for consistency and gas savings.

File: src/multiproof/tee/NitroEnclaveVerifier.sol [hunk 4]
```diff
@@ -0,0 +1,660 @@
...
+        ZkCoProcessorType,
+        bytes32,
+        bytes32,
+        bytes calldata
+    )
+        external
+        pure
+        returns (VerifierJournal[] memory)
+    {
+        revert("Not implemented");
```
Comment (leopoldjoy) [inline src/multiproof/tee/NitroEnclaveVerifier.sol]: Why must we follow the interface if we don't need this functionality? If we do need to follow it, then we should add `@notice NatSpec` documenting that these are intentionally disabled and why.

File: src/multiproof/tee/NitroEnclaveVerifier.sol [hunk 5]
```diff
@@ -0,0 +1,660 @@
+//SPDX-License-Identifier: Apache2.0
```
Comment (leopoldjoy) [inline src/multiproof/tee/NitroEnclaveVerifier.sol]: Do we need to stick with Apache-2.0 here?

File: test/multiproof/NitroEnclaveVerifier.t.sol
Comment (leopoldjoy) [inline test/multiproof/NitroEnclaveVerifier.t.sol:1]: The following revert paths have no test coverage:
- `Unknown_Zk_Coprocessor()` error (invalid coprocessor type)
- `ZkVerifierNotConfigured` error (no verifier set, no route match)
- `ZkRouteFrozen` during an actual verify() call (vs. just the `getZkVerifier` view function)
- Owner-only checks on `updateVerifierId`, `updateAggregatorId`, `removeVerifierId`, `removeAggregatorId`, `addVerifyRoute`, `freezeVerifyRoute`

File: src/multiproof/tee/NitroEnclaveVerifier.sol [hunk 6]
```diff
@@ -0,0 +1,660 @@
...
+        internal
+        view
+    {
+        // Resolve the verifier address (route-specific or default)
+        address verifier = _resolveZkVerifier(zkCoprocessor, proofBytes);
+
+        if (zkCoprocessor == ZkCoProcessorType.RiscZero) {
+            IRiscZeroVerifier(verifier).verify(proofBytes, programId, sha256(output));
+        } else if (zkCoprocessor == ZkCoProcessorType.Succinct) {
+            ISP1Verifier(verifier).verifyProof(programId, output, proofBytes);
```
Comment (leopoldjoy) [inline src/multiproof/tee/NitroEnclaveVerifier.sol:692]: We want the system to support both RiscZero and SP1 out of the box, correct? If so, then we should also add code path coverage for SP1 into the tests file below as it currently has zero test coverage.

File: test/multiproof/NitroEnclaveVerifier.t.sol [hunk 7]
```diff
@@ -0,0 +1,614 @@
...
+    bytes32 public constant ROOT_CERT = keccak256("root-cert");
+    bytes32 public constant INTERMEDIATE_CERT_1 = keccak256("intermediate-cert-1");
+    bytes32 public constant INTERMEDIATE_CERT_2 = keccak256("intermediate-cert-2");
+    bytes32 public constant VERIFIER_ID = keccak256("verifier-id");
+    bytes32 public constant AGGREGATOR_ID = keccak256("aggregator-id");
+    bytes32 public constant VERIFIER_PROOF_ID = keccak256("verifier-proof-id");
+
+    uint64 public constant MAX_TIME_DIFF = 3600; // 1 hour
+
+    function setUp() public {
```
Comment (leopoldjoy) [inline test/multiproof/NitroEnclaveVerifier.t.sol:37]: Nit: Tests run at default Foundry `block.timestamp` (1), which is unrealistic for timestamp validation logic. Consider adding `vm.warp(1_700_000_000)` or similar in `setUp()` to test with realistic timestamps.

File: src/multiproof/tee/NitroEnclaveVerifier.sol (hunk 4 above)
Comment (roger-bai-coinbase) [reply src/multiproof/tee/NitroEnclaveVerifier.sol]: Would it be better to make our own interface then? Unsure how we want to handle this

File: src/multiproof/tee/NitroEnclaveVerifier.sol [hunk 8]
```diff
@@ -0,0 +1,732 @@
+// SPDX-License-Identifier: Apache2.0
+pragma solidity ^0.8.0;
+
+import { Ownable } from "@solady/auth/Ownable.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {
+    INitroEnclaveVerifier,
```
Comment (leopoldjoy) [inline src/multiproof/tee/NitroEnclaveVerifier.sol:7]: [`TEEProverRegistry`](https://github.com/base/contracts/blob/main/src/multiproof/tee/TEEProverRegistry.sol#L11) and [`MockDevTEEProverRegistry`](https://github.com/base/contracts/blob/main/src/multiproof/mocks/MockDevTEEProverRegistry.sol#L6) still import `INitroEnclaveVerifier` / `VerifierJournal` / `ZkCoProcessorType` from the upstream lib, while this contract implements the custom interface. These are distinct Solidity types — we should probably update those contracts to use the custom interface.

File: interfaces/multiproof/tee/INitroEnclaveVerifier.sol [hunk 9]
```diff
@@ -0,0 +1,410 @@
...
+
+    /**
+     * @dev Sets the proof submitter address
+     * @param _proofSubmitter The address of the proof submitter
+     *
+     * Requirements:
+     * - Only callable by contract owner
+     * - Address must not be zero
+     */
+    function setProofSubmitter(address _proofSubmitter) external;
```
Comment (leopoldjoy) [inline interfaces/multiproof/tee/INitroEnclaveVerifier.sol:263]: Nit: `setProofSubmitter` is in the interface but the `proofSubmitter()` getter is not, so consumers using the interface type can set it but not read it. Let's also add `function proofSubmitter() external view returns (address);` to the query section?

**Review by leopoldjoy — APPROVED** [review]
